mne/dipole.py

```diff
@@ -404,14 +416,25 @@ def constraint(rd):
     # Compute the dipole moment at the final point
     Q, gof, residual = _fit_Q(fwd_data, whitener, proj_op, B, B2, B_orig,
                               rd_final)
-    amp = np.sqrt(np.sum(Q * Q))
-    norm = 1 if amp == 0 else amp
+    amp = np.sqrt(np.dot(Q, Q))
+    norm = 1. if amp == 0. else amp
     ori = Q / norm
+
+    msg = '---- Fitted : %7.1f ms' % (1000. * t)
+    if surf is not None:
+        dist_to_inner_skull = _compute_nearest(surf['rr'],
+                                               rd_final[np.newaxis, :],
+                                               return_dists=True)[1][0]
+        msg = (", distance to inner skull : %2.4f mm"
+               % (dist_to_inner_skull * 1000.))
```

Comment:
doesn't this need to be `+=`, I think this will get rid of the original part of the msg